File: Optional5/opt1.cpp
#include <stdio.h>
int stringCompare(char s1[], char s2[]);
int main(){
    char str1[50], str2[50];
    puts("Please input the first string: ");
    fgets(str1, 50, stdin);
    puts("Please input the second string: ");
    fgets(str2, 50, stdin);
    printf("The comparison result of stringCompare is %d.\n", stringCompare(str1, str2));
    return 0;
}
int stringCompare(char s1[], char s2[]){
    int i = 0;
    while (s1[i] != '\0' || s2[i] != '\0'){
        if (s1[i] != s2[i]){
            if (s1[i] > s2[i])
                return 1;
            else
                return -1;
        }
        i++;
    }
    return 0;
}

File: L8/opt1.java
import java.util.Scanner;
public class opt1{
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        System.out.println("Please input the first string: ");
        String str1 = scan.next();
        System.out.println("Please input the second string: ");
        String str2 = scan.next();
        System.out.println("The return value is " + stringCompare(str1, str2));
    }
    private static int stringCompare(String s1, String s2){
        for (int i = 0; i < s1.length(); i++){
            if (s1.charAt(i) > s2.charAt(i)){
                return 1;
            }
            else if (s1.charAt(i) < s2.charAt(i)){
                return -1;
            }
        }
        if (s1.length() < s2.length())
            return -1;
        return 0;
    }
}